### PR TITLE
[23.05] picks from main branch

### PR DESCRIPTION
**Pull request by dangowrt**
Pick updates, fixes and compatible additions from main branch openwrt-23.05 branch

### CODE REVIEW

File: include/prereq-build.mk [hunk 1]
```diff
@@ -175,23 +175,21 @@ $(eval $(call SetupHostCommand,install,Please install GNU 'install', \
 $(eval $(call SetupHostCommand,perl,Please install Perl 5.x, \
 	perl --version | grep "perl.*v5"))
 
-$(eval $(call SetupHostCommand,python,Please install Python >= 3.6, \
+$(eval $(call SetupHostCommand,python,Please install Python >= 3.7, \
```
Comment (hauke) [inline include/prereq-build.mk:178]: I would prefer if we keep python 3.6 support in 23.05.

Comment (dangowrt) [reply include/prereq-build.mk]: Ack. Removed the commit dropping support for Python 3.6.

**Comment by guidosarducci** [comment]
@dangowrt PR https://github.com/openwrt/openwrt/pull/12669 also targeted kernel 5.15 and 23.05, but was merged after the branch. Could you include this in your roll-up as well? Many thanks.

**Comment by 1715173329** [comment]
Please additionally revert this commit c07038da27cefa5a93e433909b9aca594386ddc1 as it's not available in kernel 5.15

File: tools/meson/Makefile [hunk 2]
```diff
@@ -1,11 +1,11 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=meson
-PKG_VERSION:=0.61.5
+PKG_VERSION:=1.1.0
```
Comment (hauke) [inline tools/meson/Makefile:4]: This needs at least Python 3.7, see #10543.
we should not backport this.

**Review by hauke — APPROVED** [review]

**Comment by guidosarducci** [comment]
@dangowrt Hi Daniel, there were a couple of PRs with old work I'd targeted for kernel 5.15 and getting into 23.05. I'd appreciate if you have a chance to look at #12688 and #12745 . Thanks very much.

**Pull request merged**